src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -1539,4 +1562,38 @@ public boolean collect(double amount, String reason) throws EconomyException {
 	public TownyWorld getWorld() {
 		return getHomeblockWorld();
 	}
+
+	/**
+	 * Increase Town Debt
+	 * 
+	 * Debt cannot increase beyond a certain threshold
+	 * (determined by the estimated value of the town)
+	 *
+	 * @param requestedNewDebtAmount The new debt requested
+	 * @param reason The reason for the debt
+	 */
+	public void increaseTownDebt(double requestedNewDebtAmount, String reason) throws EconomyException {
+		double actualNewDebtAmount;
+		double maximumDebtAmount = getEstimatedValueOfTown();
+
+		if(requestedNewDebtAmount + getDebtAccount().getHoldingBalance() < maximumDebtAmount) {
+			actualNewDebtAmount = requestedNewDebtAmount;
+		} else {
+			actualNewDebtAmount = maximumDebtAmount - getDebtAccount().getHoldingBalance();
+		}
+
+		getDebtAccount().collect(actualNewDebtAmount, reason);
+	}
+
+	private double getEstimatedValueOfTown() {
+		return TownySettings.getNewTownPrice() + (getTownBlocks().size() * TownySettings.getClaimPrice());
+	}
+
+	public boolean isBankrupt() {
+		try {
+			return getDebtAccount().getHoldingBalance() > 0;
```

Comment:
change to `return TownySettings.isTownBankruptcyEnabled() && getDebtAccount().getHoldingBalance() > 0;